fork-off/src/jsonrpc_client.rs

```diff
@@ -71,6 +89,46 @@ impl Client {
         (receiver, self.do_stream_all_keys(sender, at.clone()))
     }
 
+    /// Returns a map representing a single child trie
+    pub async fn get_child_storage_for_key(
+        &self,
+        child_key: StorageKey,
+        at: &BlockHash,
+    ) -> RpcResult<ChildStorageMap> {
+        let empty_prefix = StorageKey::new("0x");
+        let mut child_storage_map = ChildStorageMap::new();
+        let mut start_key = None;
+
+        loop {
+            let keys = self
+                .client
+                .get_child_keys_paged(
+                    child_key.clone(),
+                    empty_prefix.clone(),
+                    CHUNK_SIZE,
+                    start_key,
+                    Some(at.clone()),
+                )
+                .await?;
+
+            let values = self
+                .client
+                .get_child_storage_entries(child_key.clone(), keys.clone(), Some(at.clone()))
```

Comment:
I'm wondering - the `get_child_storage_entries` might be called with up to `CHUNK_SIZE` keys, can it happen that the amount of data returned here would be too much and the query would fail?

Reply:
Hmm... it's still hard bound by `CHUNK_SIZE` * (what can fit in a block), but yeah, I'm not sure if that's good enough, will change it just to be sure

Reply:
When you say _what can fit in a block_ do you mean block's maximum size (in bytes) or its maximum gas? The reason I'm asking is that, in theory, you can have a very small transaction that will write lots of keys (child storage entries?) - just write new keys in a loop until you run out of gas.

Of course, this is not very practical example but there are other cases where a simple transaction can write a lot of keys.

On the other hand, maybe it's not worthwhile to spend too much time on this since it's a testing tool that won't affect any running system and we can fix as we go and encounter this problem.

**tl;dr;**: I think I'm fine leaving this _as-is_ and deal with it if we ever have to.

Reply:
I mean size, it's not possible to write more data under a key than what fits in a block (right?)
Not sure how a lots of keys can hurt us here - we have 1:1 for (keys in child trie) : (child trie storage entries)

Reply:
Right, right, for some reason I didn't consider the case in which transaction is small but runtime/some contract generates large "value" to store it under some key

Reply:
But again, I don't think we _have to_ handle that case here right away. I think it will be most instructive to simply run this on already-existing networks and see whether it works as expected and handle the problems as they go.